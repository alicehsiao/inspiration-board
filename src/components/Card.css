.card {
  /* background-color: #F4FF81; */
  background-color: #FFDED6;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns:1fr 5fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  align-self: center;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

button {
  /* background-color: #424041; */
  background-color: #FFDED6;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  font-size: 15px;
  border-radius: 7px;
  height: 27px;
  width: 27px;
  vertical-align:middle;
  display:table-cell;
}

button:hover {
  cursor: pointer;
  color: #424041;
  background: #FFDED6;
  border: .5px black solid;
}

button:focus {
  outline:0;
}

button:focus {
  background: #FFDED6;
  outline: none;
  -webkit-box-shadow: inset 0px 0px 5px #c1c1c1;
     -moz-box-shadow: inset 0px 0px 5px #c1c1c1;
          box-shadow: inset 0px 0px 5px #c1c1c1;
}
