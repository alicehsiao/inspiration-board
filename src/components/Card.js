import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  onDelete = (event) => {
    event.preventDefault();
    this.props.onDeleteCallback(this.props.id);
  }

  render() {
    const { text } = this.props;

    return (
      <div className="card">
        <section className="card__content">
          <section className="card__content-text">
            {text}
          </section>
          <section className="card__content-emoji">
            {this.props.emoji && emoji.getUnicode(this.props.emoji)}
          </section>
          <section className="card__delete">
            <button onClick={this.onDelete}>Delete</button>
          </section>
        </section>
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,
  onDeleteCallback: PropTypes.func.isRequired,
};

export default Card;
