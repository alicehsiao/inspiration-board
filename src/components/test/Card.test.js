import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('card', () => {
  it('will match the Card Snapshot', () => {
    const wrapper = shallow(<Card
             text="My favorite winter activity is going back inside where it is warm."
             emoji="woozy"
             id={1}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
