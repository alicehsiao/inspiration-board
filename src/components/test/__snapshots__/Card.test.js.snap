// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the Card Snapshot 1`] = `
<div
  className="card"
>
  <section
    className="card__content"
  >
    <section
      className="card__content-text"
    >
      My favorite winter activity is going back inside where it is warm.
    </section>
    <section
      className="card__content-emoji"
    >
      ðŸ¥´
    </section>
  </section>
</div>
`;
